app/main/RTMediaUploadTerms.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Author: Malav Vasita <malav.vasita@rtcamp.com> *
+ */
+class RTMediaUploadTerms {
+
+	function __construct() {
+		$this->load_translation();
+		// enqueue js and css files
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts_styles' ), 999 );
+
+		// filter into uploader to show checkbox
+		add_filter( 'rtmedia_uploader_before_start_upload_button', array(
+			$this,
+			'show_terms_and_service_checkbox',
+		), 199, 1 );
+		// Filter into activity uploader to show checkbox
+		add_filter( 'rtmedia_uploader_after_activity_upload_button', array(
+			$this,
+			'show_terms_and_service_checkbox_activity',
+		), 199, 1 );
+	}
+
+	/**
+	 * loads translation
+	 */
+	public function load_translation() {
+		load_plugin_textdomain( 'rtmedia', false, basename( RTMEDIA_PATH ) . '/languages/' );
+	}
+
+	/**
+	 * loads styles and scripts
+	 * @global type $rtmedia
+	 */
+	function enqueue_scripts_styles() {
+		global $rtmedia;
+
+		$suffix = ( function_exists( 'rtm_get_script_style_suffix' ) ) ? rtm_get_script_style_suffix() : '.min';
+
+		if ( ! ( isset( $rtmedia->options ) && isset( $rtmedia->options['styles_enabled'] ) && $rtmedia->options['styles_enabled'] == 0 ) ) {
+			wp_enqueue_style( 'rtmedia-upload-terms-main', RTMEDIA_URL . 'app/assets/css/rtm-upload-terms' . $suffix . '.css', '', RTMEDIA_VERSION );
+		}
+		wp_enqueue_script( 'rtmedia-upload-terms-main', RTMEDIA_URL . 'app/assets/js/rtm-upload-terms' . $suffix . '.js', array( 'jquery' ), RTMEDIA_VERSION, true );
+		wp_localize_script( 'rtmedia-upload-terms-main', 'rtmedia_upload_terms_check_terms_message', apply_filters( 'rtmedia_upload_terms_check_terms_message', esc_html__( $rtmedia->options['general_upload_terms_error_message'], 'buddypress-media' ) ) );
+		wp_localize_script( 'rtmedia-upload-terms-main', 'rtmedia_upload_terms_check_terms_default_message', apply_filters( 'rtmedia_upload_terms_check_terms_default_message', esc_html__( 'Please check Terms of Service.', 'buddypress-media' ) ) );
```

Comment:
Use `esc_js` instead of `esc_html` here. 
Also keep it like: `esc_js( apply_filters( 'rtmedia_upload_terms_check_terms_message', __( $rtmedia->options['general_upload_terms_error_message'], 'buddypress-media' ) ) )`. 

This will also escape output coming from overridden value. @malavasita